rclrs/src/context.rs

```diff
@@ -118,6 +127,47 @@ impl Context {
     }
 }
 
+/// Returns non-ROS arguments held by context.
+///
+/// This function should be called after rcl_init. `rcl_arguments` should be `global_arguments`
+/// field of initialized context. `args` is array of input arguments passed to node.
+fn get_non_ros_arguments(
+    rcl_arguments: *const rcl_arguments_t,
+    args: &[*const c_char],
+) -> Result<Vec<String>, RclrsError> {
+    // SAFETY: In case of error -1 will be returned and handled a line below.
+    let args_status = unsafe { rcl_arguments_get_count_unparsed(rcl_arguments) };
+    if args_status == -1 {
+        return Err(RclrsError::RclError {
+            code: InvalidArgument,
+            msg: None,
+        });
+    }
+    // SAFETY: All possible negative args_status values handled above.
+    let args_num = args_status as usize;
+    if args_num == 0 {
+        return Ok(Vec::new());
+    }
+    let mut out_ptr: *mut i32 = null_mut();
+    unsafe {
+        // SAFETY: No preconditions for this function.
+        let allocator = rcutils_get_default_allocator();
+        // SAFETY: Error will be returned in case of broken rcl_arguments, no precondition for the rest.
+        rcl_arguments_get_unparsed(rcl_arguments, allocator, &mut out_ptr).ok()?;
+    }
+    // SAFETY: out_ptr won't be allocated in case of error and this code won't be reached.
+    // Indices of non-ROS args in input args array
+    let indices: Vec<i32> = unsafe { Vec::from_raw_parts(out_ptr, args_num, args_num) };
+    let mut non_ros_args: Vec<String> = Vec::with_capacity(args_num);
+
+    for i in indices.iter() {
+        // SAFETY: args are expected to have been transformed from String. Indices are expected to be positive.
```

Comment:
I don't think this is necessarily a safe assumption - I believe most arguments provided to the program as OSString, though I may be mistaken on that front.